lib/api/statistics.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime, timedelta
+
+from rest_framework import status
+from rest_framework import generics
+from django.db.models import Count, QuerySet
+from django.utils import timezone
+
+from exercise.submission_models import Submission
+from .serializers import StatisticsSerializer
+
+
+class BaseStatisticsView(generics.RetrieveAPIView):
+    """
+    Returns submission statistics for the entire system, over a given time window.
+
+    Returns the following attributes:
+
+    - `submission_count`: total number of submissions.
+    - `submitters`: number of users submitting.
+
+    Operations
+    ----------
+
+    `GET /statistics/`:
+        returns the statistics for the system.
+
+    - URL parameters:
+        - `endtime`: date and time in ISO 8601 format indicating the end point
+          of time window we are interested in. Default: now.
+        - `starttime`: date and time in ISO 8601 format indicating the start point
+          of time window we are interested in. Default: one day before endtime
+    """
+    serializer_class = StatisticsSerializer
+    endtime = timezone.now()
+    starttime = endtime - timedelta(days=1)
+
+    def get_queryset(self) -> QuerySet:
+        queryset = Submission.objects.all()
+
+        try:
+            endtime = self.request.query_params.get('endtime')
+            if endtime:
+                self.endtime = datetime.fromisoformat(endtime)
+
+            starttime = self.request.query_params.get('starttime')
+            if starttime:
+                self.starttime = datetime.fromisoformat(starttime)
```

Comment:
The instance variables `self.endtime` and `self.starttime` must be initialized here with the default values if the query parameters are empty. The instance variables are needed in the other method `get_object()`.